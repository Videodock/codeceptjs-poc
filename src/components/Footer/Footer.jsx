import React from 'react';
import { useTranslation } from 'react-i18next';
import { makeStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import images from '../../theme/images';
import styles from './Footer.styles';

const useStyles = makeStyles(styles);

export default function Footer () {
  const { t }   = useTranslation('footer');
  const classes = useStyles();

  return (
    <footer>
      <Grid
        className={classes.footer}
        direction="row"
        justify="center"
        alignItems="center"
        container
      >
        <Grid xs={12} item>
          <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/">
            <img className={classes.logo} src={images.logoAsSvg} alt={t('trompa_logo')} />
          </a>
        </Grid>
        <Grid xs={12} item>
          <Grid
            className={classes.linksRow}
            direction="row"
            justify="center"
            alignItems="flex-start"
            container
          >
            <Grid xs={12} sm={4} item>
              <Typography variant="h3">
                {t('about')}
              </Typography>

              <Grid
                justify="flex-start"
                alignItems="flex-start"
                container
              >
                <Grid className={classes.linksBlock} xs={12} md={6} item>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/about">{t('about_us')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/partners">{t('partners')}</a>
                </Grid>
                <Grid className={classes.linksBlock} xs={12} md={6} item>
                  <a target="_blank" rel="noopener noreferrer" href="https://videodock.atlassian.net/servicedesk/customer/portal/6">{t('support')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/news">{t('news')}</a>
                </Grid>
              </Grid>
            </Grid>
            <Grid xs={12} sm={4} item>
              <Typography variant="h3">
                {t('use_cases')}
              </Typography>
              <Grid
                justify="flex-start"
                alignItems="flex-start"
                container
              >
                <Grid className={classes.linksBlock} xs={12} lg={6} item>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/music-scholars">{t('music_scholars')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/choir-singers">{t('choir_singers')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/content-owners">{t('content_owners')}</a>
                </Grid>
                <Grid className={classes.linksBlock} xs={12} lg={6} item>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/instrumental-players">{t('instrumental_players')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/music-enthusiasts">{t('music_enthusiasts')}</a>
                </Grid>
              </Grid>
            </Grid>
            <Grid className={classes.linksLastColumn} xs={12} sm={4} item>
              <Typography variant="h3">
                {t('get_involved')}
              </Typography>
              <Grid
                justify="flex-start"
                alignItems="flex-start"
                container
              >
                <Grid className={classes.linksBlock} xs={12} lg={6} item>
                  <a target="_blank" rel="noopener noreferrer" href="https://trompamusic.eu/newsletter">{t('newsletter')}</a>
                  <a target="_blank" rel="noopener noreferrer" href="https://twitter.com/TrompaMusic">{t('twitter')}</a>
                </Grid>
                <Grid className={classes.linksBlock} xs={false} lg={6} item />
              </Grid>
            </Grid>
          </Grid>
        </Grid>
        <Grid xs={12} item>
          <Grid
            direction="row"
            justify="flex-start"
            alignItems="center"
            className={classes.euBlock}
            container
          >
            <Grid item>
              <img className={classes.euFlag} src={images.euFlag} alt={t('eu_flag')} />
            </Grid>
            <Grid sm={10} item>
              <Typography >
                {t('eu_flag_paragraph')}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </footer>
  );
}
