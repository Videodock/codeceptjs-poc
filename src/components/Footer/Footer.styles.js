import { createStyles } from '@material-ui/styles';

export default ({ spacing, typography, palette, breakpoints }) => createStyles({
  footer: {
    boxShadow: '0 2px 6px 0px rgba(0,0,0,0.2)',
    padding  : '48px 96px',
    '& h3'   : {
      marginBottom: spacing(3),
    },
    '& p': {
      maxWidth: 1100,
    },
    [breakpoints.only('xs')]: {
      padding: '36px 16px',
      '& h3' : {
        marginTop: spacing(2),
      },
    },
  },
  logo: {
    width                   : 138,
    marginBottom            : 48,
    [breakpoints.down('sm')]: {
      marginBottom: spacing(4),
    },
  },
  linksRow: {
    marginBottom: spacing(5),
  },
  linksLastColumn: {
    paddingLeft             : spacing(4),
    [breakpoints.down('sm')]: {
      paddingLeft: 0,
    },
  },
  linksBlock: {
    display      : 'flex',
    flexDirection: 'column',
    '& a'        : {
      fontFamily    : typography.fontFamilyOpenSans,
      fontSize      : typography.pxToRem(15.8),
      letterSpacing : 0.5,
      textDecoration: 'none',
      color         : palette.text.darkGrey,
      marginBottom  : spacing(2),
    },
  },
  euBlock: {
    [breakpoints.down('md')]: {
      justifyContent: 'flex-start',
      alignItems    : 'flex-start',
    },
  },
  euFlag: {
    width                   : 120,
    height                  : 80,
    marginRight             : spacing(3),
    [breakpoints.down('md')]: {
      alignItems  : 'flex-start',
      paddingTop  : spacing(),
      marginBottom: spacing(2),
    },
    [breakpoints.down('sm')]: {
      width : 240,
      height: 160,
    },
    [breakpoints.only('x')]: {
      paddingTop: 0,

    },
  },
});
