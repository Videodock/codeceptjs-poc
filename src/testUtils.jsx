import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
