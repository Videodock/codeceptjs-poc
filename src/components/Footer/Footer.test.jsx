import React from 'react';
import { render } from '../../testUtils';
import Footer from './Footer';

describe('<Footer />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
