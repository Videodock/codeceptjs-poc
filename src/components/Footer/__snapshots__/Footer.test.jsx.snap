// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders and matches snapshot 1`] = `
<footer>
  <div
    class="MuiGrid-root makeStyles-footer-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <a
        href="https://trompamusic.eu/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Trompa logo"
          class="makeStyles-logo-2"
          src="[object Object]"
        />
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root makeStyles-linksRow-3 MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            About
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
          >
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <a
                href="https://trompamusic.eu/about"
                rel="noopener noreferrer"
                target="_blank"
              >
                About us
              </a>
              <a
                href="https://trompamusic.eu/partners"
                rel="noopener noreferrer"
                target="_blank"
              >
                Partners
              </a>
            </div>
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
            >
              <a
                href="https://videodock.atlassian.net/servicedesk/customer/portal/6"
                rel="noopener noreferrer"
                target="_blank"
              >
                Support
              </a>
              <a
                href="https://trompamusic.eu/news"
                rel="noopener noreferrer"
                target="_blank"
              >
                News
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Use cases
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
          >
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
            >
              <a
                href="https://trompamusic.eu/music-scholars"
                rel="noopener noreferrer"
                target="_blank"
              >
                Music scholars
              </a>
              <a
                href="https://trompamusic.eu/choir-singers"
                rel="noopener noreferrer"
                target="_blank"
              >
                Choir singers
              </a>
              <a
                href="https://trompamusic.eu/content-owners"
                rel="noopener noreferrer"
                target="_blank"
              >
                Content owners
              </a>
            </div>
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
            >
              <a
                href="https://trompamusic.eu/instrumental-players"
                rel="noopener noreferrer"
                target="_blank"
              >
                Instrumental players
              </a>
              <a
                href="https://trompamusic.eu/music-enthusiasts"
                rel="noopener noreferrer"
                target="_blank"
              >
                Music enthusiasts
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-linksLastColumn-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Get involved
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
          >
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6"
            >
              <a
                href="https://trompamusic.eu/newsletter"
                rel="noopener noreferrer"
                target="_blank"
              >
                Newsletter
              </a>
              <a
                href="https://twitter.com/TrompaMusic"
                rel="noopener noreferrer"
                target="_blank"
              >
                Twitter
              </a>
            </div>
            <div
              class="MuiGrid-root makeStyles-linksBlock-5 MuiGrid-item MuiGrid-grid-lg-6"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root makeStyles-euBlock-6 MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <img
            alt="European flag"
            class="makeStyles-euFlag-7"
            src="eu_flag.jpg"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-10"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            This project has received funding from the European Union's Horizon 2020 research and innovation programme H2020-EU.3.6.3.1. - Study European heritage, memory, identity, integration and cultural interaction and translation, including its representations in cultural and scientific collections, archives and museums, to better inform and understand the present by richer interpretations of the past under grant agreement No 770376.
          </p>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
